### ISSUE TITLE
accuracy stays 0 while training, and the top_actions are [4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..., 0, 0]

### ISSUE THREAD

**Issue by misaki-sysu** [truthless11/HRL-RE#5]
Train the model with
python main.py --lr 2e-5 --datapath ../data/NYT10/

Dev and test set F1 are 0， training accuracy is 0 on each training epoh while the loss is continuing decline
After training 2 epochs, all sentences' top_actions are[4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ..., 0, 0]

Environment:
Python 3.5.2
Pytorch 1.0.1(Have change the code to Pytorch1.0 and no error came out yet )
@truthless11

**Comment by truthless11** [comment]
Did you pretrain the model first? e.g.

`python3 main.py --datapath ../data/NYT10/ --pretrain True
`

Please refer to #1 for the results of pretraining

**Comment by misaki-sysu** [comment]
@truthless11  this result I mention above had no pretrain. I had pretrained it and the accuracy , F1 score stayed 1 while pretraining, is it normal?

**Comment by truthless11** [comment]
Yes, it is. When pretraining, the output of training data is useless as it will always print out 1.0 F1 score. Please be patient and wait for the output of test data.

**Comment by misaki-sysu** [comment]
Thanks! I will wait for the pretrain result.
But, forgive my ignorance, why can't I just train the model without pretraining ?

**Comment by truthless11** [comment]
@misaki-sysu  RL suffers from the cold start problem in a large-scale state or action space so that it can hardly learn from the sparse reward, especially for hierarchical RL in such a complex task. So pretraining can prevent the agent from exploring many unreasonable cases (e.g. multiple source entities in one relational triple) thus making the training possible.

**Comment by misaki-sysu** [comment]
Thanks for you answer!
